packages/astro/src/assets/fonts/vite-plugin-fonts.ts

```diff
@@ -357,7 +409,7 @@ export function fontsPlugin({ settings, sync, logger }: Options): Plugin {
 				throw new AstroError(AstroErrorData.UnknownFilesystemError, { cause: e });
 			}
 			await Promise.all(
-				Array.from(hashToUrlMap!.entries()).map(async ([hash, url]) => {
+				Array.from(hashToUrlMap?.entries() ?? []).map(async ([hash, url]) => {
```

Comment:
I think we should check if we entries first, **then** call `Promise.all`. Classic `if` statement. 

The current code forces us to create a promise even when we don't have any entries